core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1360,6 +1472,29 @@ class ConnectionQuotas(config: KafkaConfig, time: Time, metrics: Metrics) extend
     }
   }
 
+  /**
+   * Calculates the delay needed to bring the observed connection creation rate to the IP limit.
+   * If the connection would cause an IP quota violation, un-record the connection
+   *
+   * @param address
+   * @param timeMs
+   * @return
+   */
+  private def recordIpConnectionMaybeThrottle(address: InetAddress, timeMs: Long): Long = {
+    val connectionRateQuota = connectionRateForIp(address)
+    val quotaEnabled = connectionRateQuota != DynamicConfig.Ip.DefaultConnectionCreationRate
```

Comment:
Perhaps it is better to declare another constant explicitly for unlimited connection rate quota.
1) If someone changes `DynamicConfig.Ip.DefaultConnectionCreationRate`, it may break code. Unless we have tests protecting this.
2) The readability of code will be better, since "default" could be anything.